lightning-c-bindings/demo.cpp

```diff
@@ -115,18 +115,21 @@ static int num_txs_broadcasted = 0; // Technically a race, but ints are atomic o
 void broadcast_tx(const void *this_arg, LDKTransaction tx) {
 	num_txs_broadcasted += 1;
 	//TODO
+	Transaction_free(tx);
 }
 
 struct NodeMonitors {
 	std::mutex mut;
 	std::vector<std::pair<LDK::OutPoint, LDK::ChannelMonitor>> mons;
-	LDKChainWatchInterfaceUtil* watch_util;
 	LDKLogger* logger;
+
+	void ConnectBlock(const uint8_t (*header)[80], uint32_t height, LDKCVec_C2Tuple_usizeTransactionZZ tx_data, LDKBroadcasterInterface broadcast, LDKFeeEstimator fee_est) {
+		std::unique_lock<std::mutex> l(mut);
+		for (auto& mon : mons) {
+			LDK::CVec_C2Tuple_TxidCVec_TxOutZZZ res = ChannelMonitor_block_connected(&mon.second, &header_2, tx_data, height, broadcast, fee_est, *logger);
+		}
+	}
 };
-void NodeMonitors_free(void *this_arg) {
```

Comment:
Why is this no longer needed?

Reply:
Its because we previously gave ownership of it to a `ManyChannelMonitor` object and allocated it on heap, but we now just keep it on the stack and so don't need to free it.